### DISCUSSION TITLE
Changing the directory a .tgt file

### DISCUSSION THREAD

**Discussion by Pokeman2003** [open-watcom/open-watcom-v2#1334] [Q&A] [answered]
I am very new to the OpenWatcom compiler and even newer to the IDE. I am trying to use a setup where I have my targets pointed to various output directories(IE: "/OUT/Client/Windows" and "/OUT/Server/Master",) but OpenWatcom puts the .tgt files in those directories. This is a bit of a problem because I'm trying to use Git on not just the source code, but also the IDE environment itself, and this forces me to set up complicated .gitignores to prevent catching the object and executable files. Is there a way for me to change the directory of the .tgt file without harming the operations of wmake?

**Comment by jmalak** [comment, marked as answer]
Open Watcom IDE is graphical tool for handling make files that use same logical hierarchy of nodes as wmake tool.
The target is description of all dependants and rules to create target. All these rules are defined in IDE configuration files.
On run-time IDE generate make file which is run in wmake tool. You can not change logical hierarchy which is dictated by wmake functionality. If you want you can define location of files/objects by options for tools which generate resulting node/file. By default IDE definition files are located together with object files and uses relative path to root. Of cause you can use absolute path for files but then you cannot simply copy target from one project to another one.

**Reply by Pokeman2003** [reply]
That is very annoying, but very well. Thank you for the information.

**Reply by jmalak** [reply]
It looks like strange but take into account that you can define multiple target for different platform to single project. By example DLL for OS/2 target and for Windows and appropriate Executable which use this DLL and only executable for DOS. You simply copy target to new one and change what is necessary. You can look on such project mswin.wpj in https://github.com/open-watcom/open-watcom-v2/blob/master/bld/src/win which define various targets for diferrent Windows (16/32-bit and WIN386 extender). It is in OW installation in subdirectory samples\win . Generaly you must create target where you compile all objects, source can be anywhere but it should be in same directory tree to be able use relative path.

**Reply by Pokeman2003** [reply]
My issue with the targets is that I wanted to store them within git, to make it extra easy to compile the target instantly.
I have since discovered this doesn't work, simply because despite the lack of absolute paths in every file I looked at, the target files fall apart if the top level directory that they all share is renamed in any way.

**Reply by jmalak** [reply]
In such case you can use ide2make utility which create regular makefiles from your project target file and you can compile target by wmake outside IDE. there is clean target that you can remove all build objects. We use these technique in OW project to check if IDE projects are building correctly by CI. You can look into such project in bld/idedemo makefiles which compile IDE project by example on Linux etc. because ide2make is not limited by IDE platform (Windows). 
You need to use relative path not absolute in IDE project. 

It is expected that you have properly setup OW before do such processing. If you are on GitHub then you can use following project for setup OW in your CI see https://github.com/open-watcom/setup-watcom by adding a few lines in your CI configuration.

Any reference to OW must be done by WATCOM environment variable, $(%WATCOM) in IDE project

If you give me reference to your project then I can review and comment your IDE project.

You can check your project for absolute path that you use ide2make to create make file and check contents where absolute path is used.

**Reply by jmalak** [reply]
in [open-watcom-v2](https://github.com/open-watcom/open-watcom-v2/tree/master)/[bld](https://github.com/open-watcom/open-watcom-v2/tree/master/bld)/[wasm](https://github.com/open-watcom/open-watcom-v2/tree/master/bld/wasm)/ide is project which is relative that can be used in any directory without change, doesn't depend where git repository is located.
